### PR TITLE
Start building against the first available EAP of 2022.3

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/Dependencies.kt [hunk 1]
```diff
@@ -71,7 +71,7 @@ fun Project.addIntellijToCompileClasspath(withGit4Idea: Boolean) {
     // better by Checker (and we don't plan to expose any part of the plugin as an API for other plugins).
     instrumentCode.set(false)
     if (withGit4Idea) {
-      plugins.set(listOf("git4idea"))
+      plugins.set(listOf("vcs-git"))
```
Comment (PawelLipski) [inline buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/Dependencies.kt:74]: All VCS plugins now have consistent `vcs-*` names, see https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1159#issuecomment-1264452270

File: build.gradle.kts [hunk 2]
```diff
@@ -22,12 +22,9 @@ apply<GradleVersionsFilterPlugin>()
 apply<VersionCatalogUpdatePlugin>()
 apply<TaskTreePlugin>()
 
-if (JavaVersion.current() != JavaVersion.VERSION_11 && JavaVersion.current() != JavaVersion.VERSION_17) {
-  throw GradleException(
-    """Project must be built with JDK 11 or 17 since:
-    1. as of v3.24, Checker Framework only supports JDK 8, 11 and 17: https://checkerframework.org/manual/#installation
-    2. codebase is Java 11-compatible, so can't be built on JDK 8"""
-  )
+// TODO (#983): use Gradle toolchain instead
```
Comment (PawelLipski) [inline build.gradle.kts:25]: Done on #1202

Comment (LukaszKontowski) [reply build.gradle.kts:25]: Yes, so - do we need this change in `build.gradle.kts` in this pull request? This check will be removed anyway after merging #1202 ?

File: CHANGE-NOTES.md [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+## v3.1.0
+- Add support for Intellij 2022.3.
```
Comment (mkondratek) [inline CHANGE-NOTES.md:6]: Added

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -39,11 +39,11 @@ Use IntelliJ IDEA Community Edition/Ultimate.
    `File > Settings > Build, Execution, Deployment > Compiler > Annotation Processors > Enable Annotation Processing`.
    Select `Obtain annotation processors from classpath` radio box.
 
-6. Set Project SDK to JDK 11 or 17: `Project Structure > Project`
+6. Set Project SDK to JDK 17: `Project Structure > Project`
```
Comment (mkondratek) [inline CONTRIBUTING.md:42]: why does it change?

File: build.gradle.kts [hunk 5]
```diff
@@ -148,7 +145,7 @@ allprojects {
     // since the one-parameter `addStringOption` doesn't seem to work, we need to add an extra
     // `-quiet`, which is added anyway by Gradle.
     (options as StandardJavadocDocletOptions).addStringOption("Xwerror", "-quiet")
-    (options as StandardJavadocDocletOptions).addStringOption("Xdoclint:all", "-quiet")
+    (options as StandardJavadocDocletOptions).addStringOption("Xdoclint:all,-missing", "-quiet")
```
Comment (mkondratek) [inline build.gradle.kts:148]: what do we need it for?

File: docker/Dockerfile [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
+FROM eclipse-temurin:17-jdk-jammy
 
 SHELL ["/bin/bash", "-euxo", "pipefail", "-c"]
 
 RUN \
   apt-get update; \
   apt-get install --no-install-recommends -y git openssh-client
 
-#Install GitHub Cli
+# Install the latest version of GitHub CLI.
```
Comment (mkondratek) [inline docker/Dockerfile:12]: If we now want to have those comments ending with `.` pls be consistent

**Review by mkondratek — CHANGES_REQUESTED** [review]
**plugin.xml:** 
`<depends>Git4Idea</depends>`
does it remain as it is?

I ve also seen a few other references to git4idea from comments (among the others ClassStructureTestSuite.java, project.rhino.js). pls make sure that they are still correct

**Comment by PawelLipski** [comment]
> plugin.xml:
<depends>Git4Idea</depends>
does it remain as it is?
I ve also seen a few other references to git4idea from comments (among the others ClassStructureTestSuite.java, project.rhino.js). pls make sure that they are still correct

Ha! I've reworked it a bit, now it's all correct, pls check in a sec

File: buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/Dependencies.kt [hunk 7]
```diff
@@ -39,7 +39,7 @@ fun Project.applyKotlinConfig() {
   }
 }
 
-fun Project.addIntellijToCompileClasspath(withGit4Idea: Boolean) {
```
Comment (PawelLipski) [inline buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/Dependencies.kt:42]: Meh lemme revert then

File: src/main/resources/META-INF/plugin.xml [hunk 8]
```diff
@@ -5,6 +5,11 @@
 
     <depends>com.intellij.modules.lang</depends>
     <depends>com.intellij.modules.platform</depends>
+    <!--
+      Note that since 2022.3:
+      * the plugin *folder* under $PRODUCT_ROOT$/plugins/ is called `vcs-git` and not `git4idea`, but
+      * the <id> of the plugin, as recorded in plugin's META-INF/plugin.xml is still `Git4Idea` - and that's what's expected here in `<depends>`
```
Comment (PawelLipski) [inline src/main/resources/META-INF/plugin.xml:11]: Unlike with `plugins.set(...)`, here `vcs-git` (i.e. folder name) isn't accepted

Comment (mkondratek) [reply src/main/resources/META-INF/plugin.xml:11]: cool. thanks for this comment

**Review by mkondratek — APPROVED** [review]
lgtm

**Review by LukaszKontowski — APPROVED** [review]

**Pull request merged**